Framework/CurveFitting/src/GSLMatrix.cpp

```diff
@@ -101,8 +101,16 @@ bool GSLMatrix::isEmpty() const { return m_data.empty(); }
 /// @param nx :: New first dimension
 /// @param ny :: New second dimension
 void GSLMatrix::resize(const size_t nx, const size_t ny) {
-  m_data.resize(nx * ny);
-  m_view = gsl_matrix_view_array(m_data.data(), nx, ny);
+  if (nx == 0 && ny == 0) {
```

Comment:
Is there an issue here if either `nx` or `ny` are `0`?

Reply:
Yes, GSL 2.3 (RHEL 7) will not accept a matrix / vector where any n = 0
In GSL >=2.4 this was relaxed so it will work

I had a look how existing code in the class dealt with this, and it appears that we always set a length of 1 and filled with zeros in this edge case.
I'm not sure why a user would have no parameters but it's something which previously threw an abort so I made our side deal with it as sanely as possible